core/src/main/scala/org/apache/spark/sql/hive/HiveClientUtil.scala

```diff
@@ -121,19 +125,53 @@ class HiveClientUtil(val sparkContext: SparkContext) extends Logging {
    * meta-store that is configured in the hive-site.xml file.
    */
   @transient
-  protected[sql] var client: HiveClient = newClient()
+  protected[sql] var client: HiveClient = newClientWithLogSetting()
 
 
   def closeCurrent(): Unit = {
     Hive.closeCurrent()
   }
 
+  private def newClientWithLogSetting(): HiveClient = {
+    val currentLevel = ClientSharedUtils.converToJavaLogLevel(LogManager.getRootLogger.getLevel)
+    try {
+      ifSmartConn(() => {
+        val props = new Properties()
+        props.setProperty("log4j.logger.DataNucleus.Datastore", "ERROR")
+        props.setProperty("log4j.logger.DataNucleus.Query", "ERROR")
+        ClientSharedUtils.initLog4J(null, props, currentLevel)
```

Comment:
In the call to ClientSharedUtils.initLog4J since null is being passed as log file name, it will always set logger properties for console. Make sure that in the connector mode if the user has set log file this call and the call below in finally does not redirect logging to console.